### PR TITLE
Clean event listener on `use-hide-on-click-outside` composable

### LINKED ISSUE #1725: Remove event listener in onInvalidate
## Description

There is a typo in `use-hide-on-click-outside` composable where instead of removing the event listener when cleaning up, we add one more:

https://github.com/WordPress/openverse-frontend/blob/aa5f58685140460414316d962a9a541e3b0e80dd/src/composables/use-hide-on-click-outside.js#L39-L41

## Additional context

I noticed it when auditing the event listeners trying to figure out where the memory leak comes from. I don't think it's the reason for the leak, but it should be removed it anyways.

### PR DESCRIPTION
**Pull request by AlexVCS**
## Fixes

Fixes #1725  by @obulat 

## Description

Removed the event listener in src/composables/use-hide-on-click-outside.js

## Checklist

- [x] My pull request has a descriptive title (not a vague title like `Update index.md`).
- [x] My pull request targets the *default* branch of the repository (`main`) or a parent feature branch.
- [x] My commit messages follow [best practices][best_practices].
- [x] My code follows the established code style of the repository.
- [ ] I added or updated tests for the changes I made (if applicable).
- [ ] I added or updated documentation (if applicable).
- [x] I tried running the project locally and verified that there are no visible errors.

[best_practices]:https://git-scm.com/book/en/v2/Distributed-Git-Contributing-to-a-Project#_commit_guidelines

## Developer Certificate of Origin

<details>
<summary>Developer Certificate of Origin</summary>

```
Developer Certificate of Origin
Version 1.1

Copyright (C) 2004, 2006 The Linux Foundation and its contributors.
1 Letterman Drive
Suite D4700
San Francisco, CA, 94129

Everyone is permitted to copy and distribute verbatim copies of this
license document, but changing it is not allowed.


Developer's Certificate of Origin 1.1

By making a contribution to this project, I certify that:

(a) The contribution was created in whole or in part by me and I
    have the right to submit it under the open source license
    indicated in the file; or

(b) The contribution is based upon previous work that, to the best
    of my knowledge, is covered under an appropriate open source
    license and I have the right under that license to submit that
    work with modifications, whether created in whole or in part
    by me, under the same open source license (unless I am
    permitted to submit under a different license), as indicated
    in the file; or

(c) The contribution was provided directly to me by some other
    person who certified (a), (b) or (c) and I have not modified
    it.

(d) I understand and agree that this project and the contribution
    are public and that a record of the contribution (including all
    personal information I submit with it, including my sign-off) is
    maintained indefinitely and may be redistributed consistent with
    this project or the open source license(s) involved.
```

</details>

### CODE REVIEW

File: src/composables/use-hide-on-click-outside.js
Comment (obulat) [inline src/composables/use-hide-on-click-outside.js:40]:
```suggestion
      onInvalidate(() => {
        document.removeEventListener('mousedown', onMouseDown)
      })
```
We need to actually add `removeEventListener` here so that we don't have any memory leaks. You can read more about it here: https://www.ditdot.hr/en/causes-of-memory-leaks-in-javascript-and-how-to-avoid-them#event-listeners

Comment (sarayourfriend) [inline src/composables/use-hide-on-click-outside.js:40]: Seems like there was a bit of a misunderstanding of the issue here :sweat_smile: The issue says that we need to remove the event listener, but only because we've typo'd this `onInvalidate` callback to call `addEventListener` _again_ for an event that's been added. What we meant to do was call `removeEventListener`. Easy mistake to make :slightly_smiling_face: 

```suggestion
      onInvalidate(() => {
        document.removeEventListener('mousedown', onMouseDown)
      })
```

**Review by sarayourfriend — COMMENTED** [review]
Thanks for the PR, and welcome to Openverse :rocket:

File: src/composables/use-hide-on-click-outside.js
Comment (sarayourfriend) [reply src/composables/use-hide-on-click-outside.js:40]: Oops, Olga got to it a second before me :slightly_smiling_face:

Comment (obulat) [reply src/composables/use-hide-on-click-outside.js:40]: I'm not sure who was first 😆

Comment (AlexVCS) [reply src/composables/use-hide-on-click-outside.js:40]: Whoops! Fixing it now, thank you!

**Review by obulat — APPROVED** [review]
Thank you for your quick PR, @AlexVCS! I am approving it, and if you undraft it, after a second approval it can be merged 🚀

**Review by sarayourfriend — APPROVED** [review]
:partying_face: Thanks again, @AlexVCS.

**Pull request merged**
